src/rpc/core_rpc_server.cpp

```diff
@@ -1980,7 +1980,7 @@ namespace cryptonote
     PERF_TIMER(on_get_quorum_state);
     bool r;
 
-    const std::shared_ptr<service_nodes::quorum_state> quorum_state = m_core.get_quorum_state(req.height);
+    const auto quorum_state = m_core.get_quorum_state(req.height);
```

Comment:
Only changed that because I needed to change the return type anyway.

I'd also argue that it improves readability: when `auto` is used I don't have to think whether it is a type conversion or simply giving a name to the returned object. It is not too difficult to look up the return type (which you have to do either way, because you don't know if the type you are assigning to is the same as the return type).

Reply:
So you do need to look up the return type to assign it, but thereafter it's clear what type it is. The other key point that leads into is the ability to glance over the code without having to figure out all the types.

Reply:
Hm, not sure I agree, but I'm happy to remove the `auto` if that is our convention.